actions/changelog-generate/action.yaml

```diff
@@ -0,0 +1,133 @@
+          echo  "totag=$latest_commit">> $GITHUB_OUTPUT 
+        fi
+
+        #from-tag
+        setfrom=""
+
+        if [[ -n "${{ inputs.old-tag }}" ]] && [[ "$tag_list" == *"${{ inputs.old-tag }}"* ]]; then
+        # tag provided & provided tag exists
+            setfrom=${{ inputs.old-tag }}
+        else
+        # no tag provided or a non-existing tag was provided 
+          if [ -n "$tag_list" ]; then 
+            # if any tag exists then set from-tag to the latest tag
+            setfrom=$(git describe --tags `git rev-list --tags --max-count=1`)
+          else
+            # if no tag exists then use the frist commit as fromtag
+            setfrom=$(git rev-list --max-parents=0 HEAD)
+          fi
+        fi
+        echo  "fromtag=$setfrom">> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: "Create changelog"
+      id: build_changelog
+      uses: mikepenz/release-changelog-builder-action@v3.7.0
+      with:
+        token: "${{ inputs.github-token }}"
+        configuration: "${{ inputs.config }}"
+        fromTag: "${{ steps.setTags.outputs.fromtag }}"
+        toTag: "${{ steps.setTags.outputs.totag }}"
+        outputFile: "${{ inputs.output }}"
+        fetchReviewers: "${{ inputs.fetchReviewers }}"
+        fetchReleaseInformation: "${{ inputs.fetchReleaseInformation }}"
+        commitMode: "${{ inputs.commitMode }}"
+
+    - name: Replace default branch with new Tag
+      if: ${{ steps.setTags.outputs.totag != inputs.new-tag }}
+      run: |
+        # Replace default branch with new tag
+        sed -i 's/\[${{ steps.setTags.outputs.totag }}\]/\[${{ inputs.new-tag }}\]/g' ${{ inputs.output }}
+        sed -i 's|releases/tag/${{ steps.setTags.outputs.totag }}|releases/tag/${{ inputs.new-tag }}|g' ${{ inputs.output }}
+
+        # Set Release date
+        tag_date=$(date '+%Y-%m-%d')
+        sed -i "s/Release Date: /Release Date: \[${tag_date}\]/g" ${{ inputs.output }}
+      shell: bash
+
+    - name: Update github Changelog
+      id: output-generator
+      run: |
+        # Reformat changes.md and add it to the existing changelog
+        echo 'single-changelog<<EOF' >> $GITHUB_OUTPUT
+        cat ${{ inputs.output }} >> $GITHUB_OUTPUT
+        echo 'EOF' >> $GITHUB_OUTPUT
+
+        # Create changelog.md in case of a first Release
+        if [ -n "${{ inputs.changelog-file }}" ]; then
+          sed -i 's/# /## /g' ${{ inputs.output }}
+          echo -e "\n" >> ${{ inputs.output }} 
+          if [ ! -f "${{ inputs.changelog-file }}" ]; then
+            echo $'# Changelog\n' >> ${{ inputs.changelog-file }}
+          fi
+
+          # Update global changelog and pass it to the ouput
+          sed -i "2i $(sed ':a;N;$!ba;s/\n/\\n/g' ${{ inputs.output }})" ${{ inputs.changelog-file }}
+
+          echo 'merged-changelog<<EOF' >> $GITHUB_OUTPUT
+          cat ${{ inputs.changelog-file }} >> $GITHUB_OUTPUT
+          echo 'EOF' >> $GITHUB_OUTPUT
+
+          rm ${{ inputs.output }} 
```

Comment:
Did you test this again? Deleting this here would erase all changes to the `CHANGELOG.md` file right?